### PR TITLE
[wip] ui: hardware dashboard: memory as percentage, not absolute value

### PR DESCRIPTION
**Pull request by vilterp**
image

Percentage makes it easier to see how close you are to being full. The node list now shows the value and the percentage. Maybe we should have both on the dashboard, but it's already getting big.

### CODE REVIEW

**Comment by nstewart** [comment]
This LGTM -- I'd love some design thinking in the future the Y axis and if this should be fixed, or scale (I can see arguments for both) cc @josueeee -- but I think the current fixed setup answers the original question fastest at a glance.

**Comment by piyush-singh** [comment]
My only nitpick here would be that it isn't easy to see how much capacity each node has. For a homogenous cluster, this isn't as problematic, but if you have some nodes with more memory than others, this chart becomes very cumbersome to read: you would need to navigate to another page to see which nodes have how much memory. Even if you do that, the only place we list memory capacity right now is in tooltips when you hover on memory utilization % in the node list on the overview page.

This PR is okay to merge on my end despite the above because:

- The current state that shows the raw bytes used number (vs a percentage) suffers from the same problem
- Seeing a percentage value makes this much more actionable since you get a sense for how close each node is to its memory capacity, even if you can't easily see that capacity

I've created a separate issue to track the problem with being able to identify capacity here: 28647.

File: pkg/server/status/runtime.go [hunk 1]
```diff
@@ -115,6 +115,12 @@ var (
 		Measurement: "RSS",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaRSSPct = metric.Metadata{
```
Comment (mberhault) [inline pkg/server/status/runtime.go]: While you're here, could you add a `sys.system_total` (or something similar) to export the total amount of memory in the system?

File: pkg/server/status/runtime.go [hunk 2]
```diff
@@ -419,6 +427,13 @@ func (rsr *RuntimeStatSampler) SampleEnvironment(ctx context.Context) {
 	goAllocated := ms.Alloc
 	goTotal := ms.Sys - ms.HeapReleased
 
+	totalMemory, err := GetTotalMemory(ctx)
+	if err != nil {
+		log.Warningf(ctx, "problem fetching total memory available to process: %s; memory percent metric will be empty.", err)
```
Comment (mberhault) [inline pkg/server/status/runtime.go]: I would skip this warning, we already have it at node startup, no reason to log it again every 10 seconds. Definitely skip setting the metric though.

**Comment by piyush-singh** [comment]
:lgtm:

File: pkg/server/status/runtime.go (hunk 1 above)
Comment (mberhault) [reply pkg/server/status/runtime.go]: uh. or `sys.mem.system`.

**Comment by vilterp** [comment]
Probably gonna leave this be.

**Pull request closed without merging**